Add test for `$size` system call

Rodrigodd opened the pull request:
This PR contains a test for the issue fixed by the PR https://github.com/chipsalliance/UHDM/pull/1098, as @alaindargelas requested in https://github.com/chipsalliance/UHDM/pull/1098#issuecomment-2745384219. This PR also updates the UHDM submodule to point to my PR branch, as requested.

I confirm the test fails before that change and works with the new changes. Took me a while to figure out how to do it, but a big enough generate block was enough, let me know if there is a better way to do that.

I also include in the commit the file `tests/SystemCallSize/SystemCallSize.log`, as generated in my machine. I am not sure if I should have done that, or this file should be generated by some CI workflow.

alaindargelas commented:
@Rodrigodd, as you can see, the regression massively fails in this PR. So there is something wrong with your local run if you can't see those failures locally. Are you updating you UHDM change in the Surelog source tree, then make Surelog, then run the tests?

Rodrigodd commented:
> Are you updating you UHDM change in the Surelog source tree, then make Surelog, then run the tests?

Yes, I make sure to do that, including deleting the UHDM intermediate artifacts after modifying the UHDM submodule, and verifying my changes were taking effect. Maybe I should have done a clean building before running the tests?

I will try running the tests again on my machine tomorrow, from a clean repo build, and I will check the GitHub workflow to see I am missing something,

Rodrigodd commented:
I tried running the tests in a way similar to the GitHub workflow:

```bash
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$PWD/install -S . -B build \
 && cmake --build build -j 24 \
 && cmake --install build \
 && cmake --build build --target UnitTests -j 24 \
 && pushd build && ctest --output-on-failure && popd \
 && rm -rf build && mv install build \
 && python3 scripts/regression.py run --uhdm-lint-filepath bin/uhdm-lint --jobs 24 --show-diffs --num_shards=6 --shard=1
```

and I was able to reproduce the CI failure. Honestly, I think I didn’t run the full test suite when I was testing my change, only the test I had just added, at least for the most part. Sorry about that.

I made a clean clone of Surelog, run the command above, and all tests passed. I noticed that when I updated the submodule in this PR, I also brought in many other unrelated commits. So I rebased my UHDM changes on top of the current UHDM submodule commit and re-ran the tests. All of them passed. I added a commit to this PR with that change.

So I guess some change in the range https://github.com/chipsalliance/UHDM/compare/7d90dd0...7a49c4e must have broken the tests.

alaindargelas commented:
@Rodrigodd , thanks for the diagnostic. Indeed one of those PRs is the troublemaker.

alaindargelas commented:
@Rodrigodd please update this PR. Basically remove your UHDM update from the PR.
I merged your UHDM change here: https://github.com/chipsalliance/Surelog/pull/4026 and reverted the particular PR that was breaking all the tests: https://github.com/chipsalliance/UHDM/pull/1100

Rodrigodd commented:
@alaindargelas Done.
